core/federated/RTI/rti_common.c

```diff
@@ -18,6 +18,20 @@ static rti_common_t* rti_common = NULL;
 // Global variables defined in tag.c:
 extern instant_t start_time;
 
+static tag_t latest_earlier_tag(tag_t tag) {
+  if (tag.time == NEVER || tag.time == FOREVER) {
+    return tag;
+  } else if (tag.time == 0 && tag.microstep == 0) {
```

Comment:
Is this condition really necessary?  Some platforms might use time == 0 for the start time (they may not have UTC clocks).  Even with UTC clocks, a tag of (0,0) is a legitimate tag.  The time refers to midnight on January 1, 1970.

Reply:
This function returns the greatest tag earlier than the given tag. And as far as I know, only `NEVER_TAG` is the tag earlier than `(0, 0)` in Lingua Franca. That's why I added this condition. 

Do we need to return `(-1, M_MAX)`, where M_MAX is the largest microstep, instead?

Reply:
Your other if branches will automatically return (-1, M_MAX), so if this is the right answer, there is no need for this branch.

You are right that we don’t use negative time values normally, but there is no fundamental reason we couldn’t.  In fact, we don’t normally use time values less than today…

Reply:
I got it. I deleted that `if` statement in [896e152](https://github.com/lf-lang/reactor-c/pull/349/commits/896e15293bd21958a124d7f73b02c455340f81d7).